### PR TITLE
 Bugfix/issue 214 revisit proton local

### LINKED ISSUE #214: revisit proton-local
_update v1: update with correct release url and system_

## Issue Description
There appears to be an issue with proton-local on macOS(or other system).

## Steps to Reproduce
I downloaded the proton-local binary from the [release page](https://github.com/timeplus-io/proton/releases).
Attempted to use the local command to execute.
Encountered the below error.

## Error Message & Stacktrace
Why do I launch the local will receive such info msg??

```
re@192 41plus % ./proton-macos-m2 local
proton local version 1.3.15.

:) Cannot load data for command line suggestions: Code: 49. DB::Exception: Thread #232414 status was not initialized. (LOGICAL_ERROR) (version 1.3.15)
    create stream v(id int)

CREATE STREAM v
(`id` int
)

Query id: fa348731-c9f3-4bce-972b-cf4bd3281fea

0 rows in set. Elapsed: 0.043 sec. 

Received exception:
Code: 2007. DB::Exception: Stream store is not ready. Unable to create stream. (CONFIG_ERROR)

:) create random stream v(id int default rand()%4) 

CREATE RANDOM STREAM v
(`id` int DEFAULT rand() % 4
)

Query id: be77d4aa-2068-433b-84d4-cdc4644ae475

Ok.

0 rows in set. Elapsed: 0.190 sec.
```

and below is my ubuntu(ee1ac35e45ad6b9d74e928f0f81a7e46464f7d93) running result
```

proton/build_debug/programs$ lldb proton-local
(lldb) target create "proton-local"
Current executable set to '/proton/build_debug/programs/proton-local' (x86_64).
(lldb) r
Process 519264 launched: '/proton/build_debug/programs/proton-local' (x86_64)
proton local version 1.3.15.

Process 519264 stopped
* thread #3, name = 'proton-local', stop reason = signal SIGABRT
    frame #0: 0x00007ffff7dae00b libc.so.6`raise + 203
libc.so.6`raise:
->  0x7ffff7dae00b <+203>: movq   0x108(%rsp), %rax
    0x7ffff7dae013 <+211>: xorq   %fs:0x28, %rax
    0x7ffff7dae01c <+220>: jne    0x7ffff7dae044            ; <+260>
    0x7ffff7dae01e <+222>: movl   %r8d, %eax
(lldb) bt
* thread #3, name = 'proton-local', stop reason = signal SIGABRT
  * frame #0: 0x00007ffff7dae00b libc.so.6`raise + 203
    frame #1: 0x00007ffff7d8d859 libc.so.6`abort + 299
    frame #2: 0x00000000339c03fb proton-local`DB::abortOnFailedAssertion(description="Thread #519268 status was not initialized") at Exception.cpp:47:5
    frame #3: 0x00000000339c057d proton-local`DB::handle_error_code(msg="Thread #519268 status was not initialized", code=49, remote=false, trace=size=15) at Exception.cpp:59:9
    frame #4: 0x00000000339c07b3 proton-local`DB::Exception::Exception(this=0x0000615000030040, msg="Thread #519268 status was not initialized", code=49, remote_=false) at Exception.cpp:70:5
    frame #5: 0x00000000339472d8 proton-local`DB::CurrentThread::get() at CurrentThread.cpp:36:15
    frame #6: 0x00000000475dd98a proton-local`DB::executeQueryImpl(begin="SELECT DISTINCT array_join(extract_all(name, '[\\\\w_]{2,}')) AS res FROM (SELECT name FROM system.functions UNION ALL SELECT name FROM system.table_engines UNION ALL SELECT name FROM system.formats UNION ALL SELECT name FROM system.table_functions UNION ALL SELECT name FROM system.data_type_families UNION ALL SELECT name FROM system.stream_settings UNION ALL SELECT name FROM system.settings UNION ALL SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate UNION ALL SELECT name FROM system.databases LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.tables LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.columns LIMIT 10000 SETTINGS _tp_internal_system_open_sesame=true) WHERE not_empty(res) SETTINGS _tp_internal_system_open_sesame=true", end="", context=std::__1::shared_ptr<DB::Context>::element_type @ 0x0000625000032100 strong=3 weak=3, internal=false, stage=Complete, istr=0x0000000000000000) at executeQuery.cpp:457:5
    frame #7: 0x00000000475dbbd4 proton-local`DB::executeQuery(query="SELECT DISTINCT array_join(extract_all(name, '[\\\\w_]{2,}')) AS res FROM (SELECT name FROM system.functions UNION ALL SELECT name FROM system.table_engines UNION ALL SELECT name FROM system.formats UNION ALL SELECT name FROM system.table_functions UNION ALL SELECT name FROM system.data_type_families UNION ALL SELECT name FROM system.stream_settings UNION ALL SELECT name FROM system.settings UNION ALL SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate UNION ALL SELECT name FROM system.databases LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.tables LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.columns LIMIT 10000 SETTINGS _tp_internal_system_open_sesame=true) WHERE not_empty(res) SETTINGS _tp_internal_system_open_sesame=true", context=std::__1::shared_ptr<DB::Context>::element_type @ 0x0000625000032100 strong=3 weak=3, internal=false, stage=Complete) at executeQuery.cpp:1154:30
    frame #8: 0x00000000498dc560 proton-local`DB::LocalConnection::sendQuery(this=0x000061b00001ff80, (null)=0x000061100007c588, query="SELECT DISTINCT array_join(extract_all(name, '[\\\\w_]{2,}')) AS res FROM (SELECT name FROM system.functions UNION ALL SELECT name FROM system.table_engines UNION ALL SELECT name FROM system.formats UNION ALL SELECT name FROM system.table_functions UNION ALL SELECT name FROM system.data_type_families UNION ALL SELECT name FROM system.stream_settings UNION ALL SELECT name FROM system.settings UNION ALL SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate UNION ALL SELECT name FROM system.databases LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.tables LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.columns LIMIT 10000 SETTINGS _tp_internal_system_open_sesame=true) WHERE not_empty(res) SETTINGS _tp_internal_system_open_sesame=true", query_parameters=size=0, query_id="", stage=2, (null)=0x0000000000000000, (null)=0x0000000000000000, (null)=false, process_progress_callback=function<void (const DB::Progress &)> @ 0x00007ffff12824b0) at LocalConnection.cpp:96:21
    frame #9: 0x000000004993888a proton-local`DB::Suggest::fetch(this=0x00007ffff5e1c100, connection=0x000061b00001ff80, timeouts=0x000061100007c588, query="SELECT DISTINCT array_join(extract_all(name, '[\\\\w_]{2,}')) AS res FROM (SELECT name FROM system.functions UNION ALL SELECT name FROM system.table_engines UNION ALL SELECT name FROM system.formats UNION ALL SELECT name FROM system.table_functions UNION ALL SELECT name FROM system.data_type_families UNION ALL SELECT name FROM system.stream_settings UNION ALL SELECT name FROM system.settings UNION ALL SELECT concat(func.name, comb.name) FROM system.functions AS func CROSS JOIN system.aggregate_function_combinators AS comb WHERE is_aggregate UNION ALL SELECT name FROM system.databases LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.tables LIMIT 10000 UNION ALL SELECT DISTINCT name FROM system.columns LIMIT 10000 SETTINGS _tp_internal_system_open_sesame=true) WHERE not_empty(res) SETTINGS _tp_internal_system_open_sesame=true") at Suggest.cpp:132:16
    frame #10: 0x000000004993f1a5 proton-local`void DB::Suggest::load<DB::LocalConnection>(this=0x000061100007c508)::'lambda'()::operator()() const at Suggest.cpp:109:17
    frame #11: 0x000000004993ee35 proton-local`decltype(std::declval<void DB::Suggest::load<DB::LocalConnection>(std::__1::shared_ptr<DB::Context const>, DB::ConnectionParameters const&, int)::'lambda'()>()()) std::__1::__invoke[abi:v15000]<void DB::Suggest::load<DB::LocalConnection>(std::__1::shared_ptr<DB::Context const>, DB::ConnectionParameters const&, int)::'lambda'()>(__f=0x000061100007c508) at invoke.h:394:23
    frame #12: 0x000000004993edfd proton-local`void std::__1::__thread_execute[abi:v15000]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void DB::Suggest::load<DB::LocalConnection>(std::__1::shared_ptr<DB::Context const>, DB::ConnectionParameters const&, int)::'lambda'()>(__t=size=2, (null)=__tuple_indices<> @ 0x00007ffff2179388) at thread:284:5
    frame #13: 0x000000004993e9b6 proton-local`void* std::__1::__thread_proxy[abi:v15000]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void DB::Suggest::load<DB::LocalConnection>(std::__1::shared_ptr<DB::Context const>, DB::ConnectionParameters const&, int)::'lambda'()>>(__vp=0x000061100007c500) at thread:295:5
    frame #14: 0x00007ffff7f75609 libpthread.so.0`start_thread + 217
    frame #15: 0x00007ffff7e8a133 libc.so.6`__clone + 67
(lldb) 
```

### PR DESCRIPTION
**Pull request by yokofly**
PR checklist:
- Did you run ClangFormat ?
- Did you separate headers to a different section in existing community code base ?
- Did you surround `proton: starts/ends` for new code in existing community code base ?

Please write user-readable short description of the changes:

### CODE REVIEW

File: src/Client/ClientBase.h [hunk 1]
```diff
@@ -140,6 +140,9 @@ class ClientBase : public Poco::Util::Application, public IHints<2, ClientBase>
     void updateSuggest(const ASTCreateQuery & ast_create);
 
 protected:
+    SharedContextHolder shared_context;
+    ContextMutablePtr global_context;
+
```
Comment (qijun-niu-timeplus) [inline src/Client/ClientBase.h:146]: Is stateless test change already in our code base?

**Review by qijun-niu-timeplus — APPROVED** [review]
LGTM

File: src/Client/ClientBase.h (hunk 1 above)
Comment (yokofly) [reply src/Client/ClientBase.h:146]: no, that's the reason i do not want to merge right now.
we are lacking tests to ensure the proton-local work.

Comment (chenziliang) [reply src/Client/ClientBase.h:146]: proton-local is like experimental and we can handle the test cases in a different PR.

**Pull request merged**